plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.xtend

```diff
@@ -0,0 +1,330 @@
+		val remainingRules = grammar.rules.fold(<AbstractRule>newArrayList()) [candidates, rule |
+  			val fqnFeatureName = rule.FQFeatureName
+			if (!processedNames.contains(fqnFeatureName) && !excludedFqnFeatureNames.contains(fqnFeatureName)) {
+				processedNames += fqnFeatureName
+				candidates += rule
+			}
+			candidates
+		]
+		
+		val superClass = grammar.getGenProposalProviderSuperClass
+		fileAccessFactory.createJavaFile(grammar.getGenProposalProviderClass, '''
+			/**
+			 * Represents a generated, default implementation of superclass {@link �superClass�}.
+			 * Methods are dynamically dispatched on the first parameter, i.e., you can override them 
+			 * with a more concrete subtype. 
+			 */
+			@SuppressWarnings("all")
+			public class �grammar.getGenProposalProviderClass.simpleName� extends �superClass� {
+			
+				�FOR assignment : assignments�
+					�assignment.handleAssignment�
+			  	�ENDFOR�
+				
+				�FOR rule : remainingRules�
+					public void complete�rule.FQFeatureName�(�EObject� model, �RuleCall� ruleCall, �
+							contentAssistContextClass� context, �ICompletionProposalAcceptorClass� acceptor) {
+						// subclasses may override
+					}
+		    	�ENDFOR�
+			}
+		''').writeTo(projectConfig.eclipsePluginSrcGen)	
+
+	}
+
+	private def StringConcatenationClient handleAssignment(Assignment assignment) {
+		// determine all assignment within 'assignment's containing parser rule
+		//  assigning the same feature, obtain their expected terminals, ...
+		val terminals = assignment.containingParserRule.containedAssignments().filter[
+			it.feature == assignment.feature
+		].map[
+			terminal
+		].toList
+
+		// ... and determine the types of those terminals 
+		val terminalTypes = terminals.map[ eClass ].toSet;
+
+		'''
+			public void complete�assignment.FQFeatureName�(�EObject� model, �Assignment� assignment, �
+					contentAssistContextClass� context, �ICompletionProposalAcceptorClass� acceptor) {
+				�IF terminalTypes.size > 1�
+					�terminals.handleAssignmentOptions�
+				�ELSE�
+					�assignment.terminal.assignmentTerminal("assignment.getTerminal()")�
+				�ENDIF�
+			}
+		'''
+	}
+	
+	private def StringConcatenationClient handleAssignmentOptions(Iterable<AbstractElement> terminals) {
+		val processedTerminals = newHashSet();
+		
+		// for each type of terminal occurring in 'terminals' (evaluated lazily) ...
+		val candidates = terminals.fold(<AbstractElement>newHashSet) [ candidates, terminal |
+			if (!processedTerminals.contains(terminal.eClass)) {
+				processedTerminals += terminal.eClass
+				candidates += terminal
+			}
+			candidates
+		]
+		
+		// ... generate an 'instanceof' clause
+		'''
+			�FOR terminal : candidates�
+				if (assignment.getTerminal() instanceof �terminal.eClass.instanceClass�) {
+					�terminal.assignmentTerminal("assignment.getTerminal()")�
+				}
+			�ENDFOR�		
+		'''
+	}
+
+	private def dispatch StringConcatenationClient assignmentTerminal(AbstractElement element, String accessor) {
```

Comment:
what about
```
.... {
val list = alternatives.elements
'''
...
```
does the following work too?
```
... '''
«val list = alternatives.elements»
«FOR ...
```
Or should I stick to the braces?  I'm not sure whether I can test that case on my own.